pkg/scheduler/framework/cycle_state_test.go

```diff
@@ -31,44 +34,98 @@ func (f *fakeData) Clone() StateData {
 	return copy
 }
 
-func TestCycleStateClone(t *testing.T) {
-	var key StateKey = "key"
-	data1 := "value1"
-	data2 := "value2"
+var key StateKey = "key"
 
-	state := NewCycleState()
-	originalValue := &fakeData{
-		data: data1,
-	}
-	state.Write(key, originalValue)
-	stateCopy := state.Clone()
+// createCycleStateWithFakeData creates *CycleState with fakeData.
+// The given data is used in stored fakeData.
+func createCycleStateWithFakeData(data string, recordPluginMetrics bool) *CycleState {
+	c := NewCycleState()
+	c.Write(key, &fakeData{
+		data: data,
+	})
+	c.SetRecordPluginMetrics(recordPluginMetrics)
+	return c
+}
 
-	valueCopy, err := stateCopy.Read(key)
-	if err != nil {
-		t.Errorf("failed to read copied value: %v", err)
-	}
-	if v, ok := valueCopy.(*fakeData); ok && v.data != data1 {
-		t.Errorf("clone failed, got %q, expected %q", v.data, data1)
+func TestCycleStateClone(t *testing.T) {
+	tests := []struct {
+		name            string
+		state           *CycleState
+		wantClonedState *CycleState // clonedState is CycleState cloned by Clone().
+	}{
+		{
+			name:            "clone with recordPluginMetrics true",
+			state:           createCycleStateWithFakeData("data", true),
+			wantClonedState: createCycleStateWithFakeData("data", true),
+		},
+		{
+			name:            "clone with recordPluginMetrics false",
+			state:           createCycleStateWithFakeData("data", false),
+			wantClonedState: createCycleStateWithFakeData("data", false),
+		},
+		{
+			name:            "clone with nil CycleState",
+			state:           nil,
+			wantClonedState: nil,
+		},
 	}
 
-	originalValue.data = data2
-	original, err := state.Read(key)
-	if err != nil {
-		t.Errorf("failed to read original value: %v", err)
-	}
-	if v, ok := original.(*fakeData); ok && v.data != data2 {
-		t.Errorf("original value should change, got %q, expected %q", v.data, data2)
-	}
+	cmpOpts := []cmp.Option{
+		cmp.AllowUnexported(CycleState{}),
+		cmp.Comparer(func(a, b *sync.Map) bool {
```

Comment:
changed `Comparer` from `sync.Map` to `*sync.Map`. 
This resolves the following go vet issue.

```
func passes lock by value: sync.Map contains sync.Mutex
```

Reply:
I'd pursue working the govet error around cuz defining a pointer of sync.Map sounds weird...

Reply:
I don't know how we can avoid the error. The comparer would receive elements by copy, and it's unsafe to copy a lock.

Reply:
Hmm, I don't have an idea to avoid the issue from go vet (other than using `*sync.Map`).

One way is to give up using `cmp.Diff` and `cmp.Equal`... 
In that case, the test would be longer, but would look like this:
https://github.com/kubernetes/kubernetes/commit/55c5d51286ff947e3c2233d92b26aa7911f6222e

@alculquicondor @Huang-Wei 
Which do you prefer, the current implementation of this branch or the implementation without `cmp.Diff` like https://github.com/kubernetes/kubernetes/commit/55c5d51286ff947e3c2233d92b26aa7911f6222e?

Reply:
My preference is for the implementation in this PR, but I won't oppose to the other if Wei feels strongly about the pointer change.

Reply:
@sanposhiho Given the `go vet` is trying to avoid copy values containing mutex. We can tweak the test like this: https://gist.github.com/Huang-Wei/5f4bd967d3adf3d738531c09214bc307#file-108727-diff.

Reply:
Yup, we can pursue debug readability over being restricted with cmp.Diff.

Reply:
👍 
Change not to use cmp.Diff in a959c3bf7c96911361f4688895494dcf7855b9dc.